examples/classic_controllers/classic_controllers.py

```diff
@@ -648,97 +865,613 @@ def __init__(self, environment, stages, ref_states, color='g', **controller_kwar
+        psi_max_ = self.u_dc / (np.sqrt(3) * omega * self.p)
+        k_i = 2 * omega * self.p / self.u_dc
+
+        i_gain = self.i_gain / k_i
+        psi_delta = i_gain * (a_delta * self.tau + self.integrated)
+
+        if self.psi_low <= psi_delta <= self.psi_high:
+            if self.limited:
+                self.integrated = self.integrated_reset
+                self.limited = False
+            self.integrated += a_delta * self.tau
+
+        else:
+            psi_delta = np.clip(psi_delta, self.psi_low, self.psi_high)
+            self.limited = True
+
+        psi = psi_max_ + psi_delta
+        if self.plot_modulation:
+            if self.k == 0:
+                self.initialize_modulation_plot()
+            self.k_list_a.append(self.k * self.tau)
+            self.a_list.append(a)
+            self.psi_delta_list.append(psi_delta)
+
+            if self.k % self.update_interval == 0:
+                    self.a_plot.scatter(self.k_list_a, self.a_list, c='tab:blue', s=3)
+                    self.psi_delta_plot.scatter(self.k_list_a, self.psi_delta_list, c='tab:blue', s=3)
+                    self.a_plot.set_xlim(max(self.k * self.tau, 1) - 1, max(self.k * self.tau, 1))
+                    self.psi_delta_plot.set_xlim(max(self.k * self.tau, 1) - 1, max(self.k * self.tau, 1))
+                    self.k_list_a = []
+                    self.a_list = []
+                    self.psi_delta_list = []
+
+        return psi
+
+    def initialize_modulation_plot(self):
+        if self.plot_modulation:
+            plt.ion()
+            self.fig_modulation = plt.figure('Modulation Controller')
+            self.a_plot = plt.subplot2grid((1, 2), (0, 0))
+            self.psi_delta_plot = plt.subplot2grid((1, 2), (0, 1))
+
+            self.a_plot.set_title('Modulation')
+            self.a_plot.axhline(self.k_ * self.a_max, c='tab:orange', label=r'$a^*$')
+            self.a_plot.plot([], [], c='tab:blue', label='a')
+            self.a_plot.set_xlabel('t / s')
+            self.a_plot.set_ylabel('a')
+            self.a_plot.grid(True)
+            self.a_plot.set_xlim(0, 1)
+            self.a_plot.legend(loc=2)
+
+            self.psi_delta_plot.set_title(r'$\Psi_\mathrm{\Delta}$')
+            self.psi_delta_plot.axhline(self.psi_low, c='tab:red', linestyle='dashed', label='Limit')
+            self.psi_delta_plot.axhline(self.psi_high, c='tab:red', linestyle='dashed')
+            self.psi_delta_plot.plot([], [], c='tab:blue', label=r'$\Psi_\mathrm{\Delta}$')
+            self.psi_delta_plot.set_xlabel('t / s')
+            self.psi_delta_plot.set_ylabel(r'$\Psi_\mathrm{\Delta} / Vs$')
+            self.psi_delta_plot.grid(True)
+            self.psi_delta_plot.set_xlim(0, 1)
+            self.psi_delta_plot.legend(loc=2)
+
+            self.a_list = []
+            self.psi_delta_list = []
+            self.k_list_a = []
+
+
+    def reset(self):
+        self.integrated = 0
```

Comment:
```suggestion
        self.integrated = self.integrated_reset
```
seems more reasonable?